[#104] Add blocking http actor service interface

ztkmkoo opened the pull request:
### motivation
We aim to full asynchronous services, so i want to make some actors to handling the blocking job.
So user(developer who uses dss) implement AbstractDssOkHttp3ClientService and add to dss server that they can use the http service.

### details
- add http client service actor implement
- add http client service message and http client service interface and test
- add abstract http client service implement for okhttp3

ztkmkoo committed:
[#104] add blocking http actor service interface

- add blocking service message interface and service interface

ztkmkoo committed:
[#104] add blocking http actor service interface

- add http client service message and http client service interface and test

ztkmkoo committed:
[#104] add blocking http actor service interface

- add http client service actor implement

ztkmkoo committed:
[#104] add blocking http actor service interface

- rename

ztkmkoo committed:
[#104] add blocking http actor service interface

- fix test error
- add okhttp3 dependency
- add abstract http client service implement for okhttp3

Amazansia commented on dss-core/src/main/java/io/github/ztkmkoo/dss/core/actor/blocking/rest/DssHttpClientServiceActor.java:
```diff
@@ -0,0 +1,39 @@
+ */
+public class DssHttpClientServiceActor<R extends DssBlockingRestCommand.HttpRequest> {
+
+    public static <R extends DssBlockingRestCommand.HttpRequest> Behavior<DssBlockingRestCommand> create(DssHttpClientService<R> service, Class<R> requestClass) {
```
I'm a little confused, is it okay to sort like "<R, variable, and constructor", not sort like "variable, constructor, and <R"?
Maybe I have to redo #36 ...

ztkmkoo replied:
Hi, it looks i miss some details bout class format... 
I'm so sorry.

i mentioned it at #36 issue...
``` java
class Foo {

    // static fields
    // static initializer
    // static methods
    // instance fields
    // constructors
    // instance methods
}
```

ztkmkoo committed:
[#104] add blocking http actor service interface

- add doc to DssHttpClientService
- add some tests

ztkmkoo committed:
[#104] add blocking http actor service interface

- fix test error

ztkmkoo committed:
[#104] add blocking http actor service interface

- add DssRestRequestCommand.java and its test

ztkmkoo committed:
[#104] add blocking http actor service interface

- after read complete http message than send DssRestRequestCommand to service actor

ztkmkoo committed:
[#104] add blocking http actor service interface

- add blocking service resolver and its implements

ztkmkoo committed:
[#104] add blocking http actor service interface

- add service property and its rest implement

ztkmkoo committed:
[#104] add blocking http actor service interface

- remove unnecessary suppress warning

ztkmkoo committed:
[#104] add blocking http actor service interface

- modify DssBlockingService interface

ztkmkoo committed:
[#104] add blocking http actor service interface

- improve blocking service resolover

ztkmkoo committed:
[#104] add blocking http actor service interface

- improve service interface for asynchronous

ztkmkoo committed:
[#104] add blocking http actor service interface

- fix rest service actors
